### ISSUE TITLE
tk dependency

### ISSUE THREAD

**Issue by richfromm** [thomasloupe/Slackord1#6]
You do **not** want to `pip install tk`. That's something called "TensorKit":
https://pypi.org/project/tk/

tkinter seems to be part of the Python standard library:
https://docs.python.org/3/library/tkinter.html

Using Python via brew on MacOS, I was initially getting the following error:
```
>>> import tkinter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'
```

This was solved via `brew install python-tk`
Courtesy of this slackoverflow post: https://stackoverflow.com/a/67465708/9797192

I **think** that using the standard MacOS python (i.e. `/usr/bin/python3`), that `tkinter` just works, nothing more needed to install.

I don't have the time right now to report back on Linux, but if you'd like I can later try this on Ubuntu (sorry, not the latest version) and see what I get.

But I have no way of testing on Windows, which is one reason I'm not just editing the README and submitting a pull request.

**Comment by richfromm** [comment]
I see that Ubuntu has a `python-tk` package as well, so if you see this there, it might be as simple as `sudo apt-get install python-tk`: https://packages.ubuntu.com/search?keywords=python-tk&searchon=names&suite=jammy&section=all

I have no ability to test for RedHat, and apparently I'd need to create a developer account even to see via a web interface if there's some `python-tk` package that could be installed via `yum`: https://unix.stackexchange.com/a/6869

**Comment by richfromm** [comment]
Okay, for a RedHat-based distribution (e.g. CentOS), stackoverflow claims you'd want `sudo yum install python3-tkinter`: https://stackoverflow.com/a/42244548/9797192

I have no way of testing this.

And yes, I realize that I'm ignoring py2 in all of this. It is 2022 already.

**Comment by richfromm** [comment]
Perhaps update the README to indicate something like tkinter might just be installed as part of your python distribution, if you see the error above (or want to explicitly test via the interactive python shell using `import tkinter`), then you need to install PythonTk, and give some suggested methods based on OS and/or Python distribution.

**Comment by thomasloupe** [comment]
Hey @richfromm, thanks for bringing this up. I think that's the best route here. Providing extremely detailed instructions is beyond the scope of the README, even though its intention is to make things easier. I'll provide some basic instructions on installing tkinter and discord.py for Mac, Linux and Windows. Thankfully, I have all three OS's here to test, so I can knock this out.

Thanks again!

**Comment by thomasloupe** [comment]
This should be updated now with correct instructions for Ubuntu, Mac, and Windows. I've tested all three on each OS with success, so I'll go ahead and close this issue out. Thanks again for bringing this to my attention.

**Comment by richfromm** [comment]
Sorry, I'm just returning to this now.

It's an improvement, but still not fully correct. `pip install tk` is **never** a relevant instruction, regardless of platform. It might not hurt, but it's not going to do anything useful, b/c it has absolutely nothing to do with tkinter.

Basically, in some cases you don't need to do anything, tkinter should just be part of your python distribution. The sign of that not being the case is if you get the `ModuleNotFoundError` error that I mentioned above. It that happens, you can solve it with:
`sudo apt-get install python-tk` for Debian-based Linux distributions (including Ubuntu)
`sudo yum install python3-tkinter` for RedHat-based Linux distributions (including CentOS)
`brew install python-tk` for Homebrew (MacOS)

The one piece I have no clue of, and no means for testing, is Windows. I'll try to search around a bit more.

Once I actually fully demonstrate this end to end, I can put up a PR with suggested edits.

**Comment by richfromm** [comment]
This suggests that tkinter is included on Windows as of Python 3.7:
https://www.activestate.com/resources/quick-reads/how-to-install-tkinter-in-windows/

I think anticipating any and all possible configurations is beyond the scope of this README, and simply saying update to a sufficiently new Python (3.7 was released in 2018, more than 4 years ago, and the latest release is currently 3.10) is sufficient.

**Comment by thomasloupe** [comment]
This is correct. I went through the installer myself and verified this for Windows and Mac (not homebrew).

Windows:
![image](https://user-images.githubusercontent.com/6563450/187075879-3d75fa91-28db-47a0-aaab-15518b64d3a4.png)

Mac:
![image](https://user-images.githubusercontent.com/6563450/187077614-8377163f-ad50-491b-aa63-1fb97df2a872.png)

RHEL/CentOS:
```
[root@orion bin]# sudo yum install -y python3 tkinter python3-pip; python3 -m pip install discord.py
Loaded plugins: fastestmirror, product-id, search-disabled-repos, subscription-manager

This system is not registered with an entitlement server. You can use subscription-manager to register.

Loading mirror speeds from cached hostfile
 * base: mirror.grid.uchicago.edu
 * epel: iad.mirror.rackspace.com
 * extras: mirror.pit.teraswitch.com
 * updates: mirror.math.princeton.edu
Package python3-3.6.8-18.el7.x86_64 already installed and latest version
Package tkinter-2.7.5-92.el7_9.x86_64 already installed and latest version
Package python3-pip-9.0.3-8.el7.noarch already installed and latest version
Nothing to do
WARNING: Running pip install with root privileges is generally not a good idea. Try `__main__.py install --user` instead.
Requirement already satisfied: discord.py in /usr/local/lib/python3.6/site-packages
Requirement already satisfied: aiohttp<3.8.0,>=3.6.0 in /usr/local/lib64/python3.6/site-packages (from discord.py)
Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: chardet<5.0,>=2.0 in /usr/local/lib/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib64/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: async_timeout<4.0,>=3.0 in /usr/local/lib/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib64/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: typing_extensions>=3.6.5 in /usr/local/lib/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: idna-ssl>=1.0 in /usr/local/lib/python3.6/site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py)
Requirement already satisfied: idna>=2.0 in /usr/local/lib/python3.6/site-packages (from yarl<2.0,>=1.0->aiohttp<3.8.0,>=3.6.0->discord.py)
[root@orion bin]#
```

For Mac and Linux, I have created a one-liner command to install python3, tkinter (if it's not already installed), and to install the discord.py pip module.
